### PR TITLE
Assistants YAML config

### PR DESCRIPTION
**Pull request by raulraja**
Given some yaml config such as:
```yaml
assistant_id: "asst_S3UbzpWIYuhsFu5cRG7b4dha"
model: "gpt-4-1106-preview"
name: "My Custom Test Assistant"
description: "A versatile AI assistant capable of conversational and informational tasks."
instructions: "This assistant is designed to provide informative and engaging conversations, answer queries, and execute code when necessary."
tools:
   - type: "code_interpreter"
   - type: "retrieval"
   - type: "function"
      name: "get_stock_price"
      description: "Get the current stock price for a given stock symbol."
      parameters: '{
              "type": "object",
              "properties": {
                "symbol": {
                  "type": "string",
                  "description": "The stock symbol"
                }
              },
              "required": [
                "symbol"
              ]
        }'
file_ids:
   - "${file.id}"
metadata: null
```
Creates or updates an Open AI assistant. 
Currently metadata is being ignored until we find the best way to represent it. Current value is `null`.
```txt
assistant: AssistantObject(id=asst_S3UbzpWIYuhsFu5cRG7b4dha, object=assistant, createdAt=1705948119, name=My Custom Test Assistant, description=A versatile AI assistant capable of conversational and informational tasks., model=gpt-4-1106-preview, instructions=This assistant is designed to provide informative and engaging conversations, answer queries, and execute code when necessary., tools=[AssistantToolsCode(type=code_interpreter), AssistantToolsRetrieval(type=retrieval), AssistantToolsFunction(function=FunctionObject(name=get_stock_price, parameters={"type":"object","properties":{"symbol":{"type":"string","description":"The stock symbol"}},"required":["symbol"]}, description=Get the current stock price for a given stock symbol.), type=function)], fileIds=[file-tofd2Y37ykgnnaownzqBdP0r], metadata={})
run: in_progress
received message: MessageContentTextObjectText(value=What does this file say?, annotations=[])
step: tool_calls
run: completed
received message: MessageContentTextObjectText(value=The file contains the following text:
Hello World!, annotations=[])
```

### CODE REVIEW

**Comment by raulraja** [comment]
It looks like only the JVM target is supported https://github.com/charleskorn/kaml/issues/232. We may need to do further research to find an alternative for multiplatform.

File: core/src/commonMain/kotlin/com/xebia/functional/xef/llm/assistants/Assistant.kt [hunk 1]
```diff
@@ -62,5 +69,96 @@ class Assistant(
...
+      return if (assistantRequest.assistantId != null) {
+        val assistant = Assistant(assistantRequest.assistantId, assistantsApi, api)
+        assistant.modify(
+          ModifyAssistantRequest(
+            name = assistantRequest.name,
+            description = assistantRequest.description,
+            instructions = assistantRequest.instructions,
+            tools = assistantTools(assistantRequest),
+            fileIds = assistantRequest.fileIds,
+            metadata = null // assistantRequest.metadata
```
Comment (javipacheco) [inline core/src/commonMain/kotlin/com/xebia/functional/xef/llm/assistants/Assistant.kt:126]: Should we manage the metadata or can we wait for addressing in another PR?

**Review by javipacheco — DISMISSED** [review]
🚀 LGTM!

Just a minor comment, but maybe it's not important for now

File: core/src/commonMain/kotlin/com/xebia/functional/xef/llm/assistants/Assistant.kt (hunk 1 above)
Comment (raulraja) [reply core/src/commonMain/kotlin/com/xebia/functional/xef/llm/assistants/Assistant.kt:162]: Let's address in another PR we don't use it yet and not sure what the best format is

File: core/src/commonMain/kotlin/com/xebia/functional/xef/llm/assistants/Assistant.kt [hunk 2]
```diff
@@ -98,5 +105,96 @@ class Assistant(
...
+            model = assistantRequest.model,
+            name = assistantRequest.name,
+            description = assistantRequest.description,
+            instructions = assistantRequest.instructions,
+            tools = assistantTools(assistantRequest),
+            fileIds = assistantRequest.fileIds,
+            metadata = null // assistantRequest.metadata
+          ),
+          assistantsApi,
+          api
```
Comment (realdavidvega) [inline core/src/commonMain/kotlin/com/xebia/functional/xef/llm/assistants/Assistant.kt:177]:
```suggestion
        Assistant(
          request = CreateAssistantRequest(
            model = assistantRequest.model,
            name = assistantRequest.name,
            description = assistantRequest.description,
            instructions = assistantRequest.instructions,
            tools = assistantTools(assistantRequest),
            fileIds = assistantRequest.fileIds,
            metadata = null // assistantRequest.metadata
          ),
          assistantsApi = assistantsApi,
          api = api
```

**Review by realdavidvega — COMMENTED** [review]
Couple of suggestions to fix the compilation issues after adding the `toolsConfig` arg.

**Review by Montagon — APPROVED** [review]

**Pull request merged**